node/out_response.js

```diff
@@ -218,10 +218,24 @@ TChannelOutResponse.prototype.sendError = function sendError(codeString, message
 TChannelOutResponse.prototype.emitFinish = function emitFinish() {
     var self = this;
 
-    if (!self.end) {
-        self.end = self.timers.now();
+    if (self.end) {
+        self.logger.warn('out response double emitFinish', {
+            end: self.end,
+            now: Date.now(),
```

Comment:
in fact hoist a `var now = self.channel.timers.now();` and de-dupe with the update call down below this error check branch